base/stat.jl

```diff
@@ -417,68 +460,107 @@ isfile(st::StatStruct) = filemode(st) & 0xf000 == 0x8000
 
 """
     islink(path) -> Bool
+    islink(path_elements...) -> Bool
 
-Return `true` if `path` is a symbolic link, `false` otherwise.
+Return `true` if `path` points to a symbolic link, `false` otherwise.
 """
 islink(st::StatStruct) = filemode(st) & 0xf000 == 0xa000
 
 """
     issocket(path) -> Bool
+    issocket(path_elements...) -> Bool
 
-Return `true` if `path` is a socket, `false` otherwise.
+Return `true` if `path` points to a socket, `false` otherwise.
 """
 issocket(st::StatStruct) = filemode(st) & 0xf000 == 0xc000
 
 # mode permission predicates
 
 """
     issetuid(path) -> Bool
+    issetuid(path_elements...) -> Bool
+    issetuid(stat_struct) -> Bool
 
-Return `true` if `path` has the setuid flag set, `false` otherwise.
+Return `true` if the file at `path` has the setuid flag set, `false` otherwise.
+Or the equivalent as indicated by the file descriptor `stat_struct`.
 """
 issetuid(st::StatStruct) = (filemode(st) & 0o4000) > 0
 
 """
     issetgid(path) -> Bool
+    issetgid(path_elements...) -> Bool
+    issetgid(stat_struct) -> Bool
 
-Return `true` if `path` has the setgid flag set, `false` otherwise.
+Return `true` if the file at `path` has the setgid flag set, `false` otherwise.
+Or the equivalent as indicated by the file descriptor `stat_struct`.
 """
 issetgid(st::StatStruct) = (filemode(st) & 0o2000) > 0
 
 """
     issticky(path) -> Bool
+    issticky(path_elements...) -> Bool
+    issticky(stat_struct) -> Bool
 
-Return `true` if `path` has the sticky bit set, `false` otherwise.
+Return `true` if the file at `path` has the sticky bit set, `false` otherwise.
+Or the equivalent as indicated by the file descriptor `stat_struct`.
 """
 issticky(st::StatStruct) = (filemode(st) & 0o1000) > 0
 
 """
-    uperm(file)
+    uperm(path)
+    uperm(path_elements...)
+    uperm(stat_struct)
+
+Get the permissions of the owner of the file at `path` as a bitfield,
+Or the equivalent as indicated by the file descriptor `stat_struct`.
 
-Get the permissions of the owner of the file as a bitfield of
 
 | Value | Description        |
 |:------|:-------------------|
 | 01    | Execute Permission |
 | 02    | Write Permission   |
 | 04    | Read Permission    |
 
-For allowed arguments, see [`stat`](@ref).
+The fact that a bitfield is returned means that if the premission
+is read+write, the bitfield is "110", which maps to the decimal
+value of 0+2+4=6. This is reflected in the printing of the
+returned `UInt8` value.
+
+```jldoctest
+julia> touch("dummy_file")  # Create test-file without contents;
+
+julia> uperm("dummy_file")
+0x06
+
+julia> bitstring(ans)
+"00000110"
+
+julia> has_read_permission(path) = bitstring(uperm(path))[end-2]=='1';
```

Comment:
```suggestion
julia> has_read_permission(path) = uperm(path) & 4 != 0;
```
I think it's better to model bit-twiddling without using `bitstring`; while performance of `bitstring` is negligible compared to the stat call, it's still IMO good form to avoid stringification.

Reply:
While I believe you make a technically good point, your change killed the pedagogical aspect for me completely, which was my whole point in adding it. I am unable to parse `uperm(path) & 4 != 0` in my head, but indexing into the bitstring, which is printed above, is trivial, and makes very clear what is going on.

Peraps it is possible to add your version as a comment? Along the lines of
```julia
has_read_permission(path) = bitstring(uperm(path))[end-2]=='1';    # More efficient, less readable:  uperm(path) & 4 != 0
```

Reply:
Would `uperm(path) & 0b00000100 != 0` be easier to parse?

Reply:
Yhea, it would. I'll commit that.